### PR TITLE
Pulp parameters

### PR DESCRIPTION
**Pull request by cristi1979**
This should make pulp module fully configurable.

Please let me know if the patch is too big. I'm not sure how I can split this one in multiple patches.

### CODE REVIEW

**Comment by ehelms** [comment]
Looks like there are some formatting updates and additional parameters being added in this PR. You could split that into two PRs and might make it easiest when I review to detect the actual parameters aspects.

**Comment by cristi1979** [comment]
Sure, I will do it. But I don't understand what the different PRs should contain.

File: manifests/config.pp [hunk 1]
```diff
@@ -1,71 +1,86 @@
...
-    }
-  }
-
-  file {'/var/lib/pulp/packages':
+  file { '/var/lib/pulp/packages':
     ensure => directory,
     owner  => 'apache',
     group  => 'apache',
-    mode   => '0755';
+    mode   => '0755',
```
Comment (ehelms) [inline manifests/config.pp]: For example, this is a formatting change.

File: manifests/init.pp [hunk 2]
```diff
@@ -3,176 +3,268 @@
...
-# $qpid_ssl_cert_db             The location of the Qpid SSL cert database
+# $https_key::                 apache private certificate for ssl
 #
-# $qpid_ssl_cert_password_file  Location of the password file for the Qpid SSL cert
+# $ssl_ca_cert::               full path to the CA certificate used to sign the Pulp
+#                              server's SSL certificate; consumers will use this to verify the
+#                              Pulp server's SSL certificate during the SSL handshake
 #
-# $user_groups::                Additional user groups to add the qpid user to
+# $user_cert_expiration::      number of days a user certificate is valid
```
Comment (ehelms) [inline manifests/init.pp]: While something like this is a functionality change.

**Comment by ehelms** [comment]
Hopefully that makes sense for splitting it into two PRs.

**Pull request closed without merging**